framework/src/actions/ActionWarehouse.C

```diff
@@ -312,6 +309,9 @@ ActionWarehouse::executeAllActions()
 void
 ActionWarehouse::executeActionsWithAction(const std::string & task)
 {
+  // Set the current task name
+  _current_task = task;
```

Comment:
This is a good change.  Was it actually required to make all of this work?

Note: I'm not criticizing, just curious...